// meli-nodejs-sdk/dist/lib/ApiClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiClient = void 0;
const superagent_1 = __importDefault(require("superagent"));
class ApiClient {
    constructor(accessToken = '') {
        this.accessToken = accessToken;
        this.baseUrl = 'https://api.mercadolibre.com';
    }
    /**
     * Calls the endpoint and return a Promise of ApiResponse with the MercadoLibre API response
     * @param {ApiRequest} apiRequest
     * @returns {Promise<ApiResponse>}
     */
    async callApi(apiRequest) {
        const request = (0, superagent_1.default)(this.baseUrl);
        request.method = apiRequest.method;
        request.url = `${this.baseUrl}${apiRequest.path}`;
        if (this.accessToken !== '') {
            request.auth(this.accessToken, { type: "bearer" });
        }
        if (apiRequest.queryParams) {
            let queryParamsString = apiRequest.queryParams.toString();
            request.url = `${request.url}?${queryParamsString}`;
        }
        await request.send();
        return {
            statusCode: (await request).statusCode,
            body: (await request).body
        };
    }
}
exports.ApiClient = ApiClient;
