// mercadolibre-nodejs-sdk/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient.default;
  }
});
Object.defineProperty(exports, "Attributes", {
  enumerable: true,
  get: function get() {
    return _Attributes.default;
  }
});
Object.defineProperty(exports, "AttributesValueStruct", {
  enumerable: true,
  get: function get() {
    return _AttributesValueStruct.default;
  }
});
Object.defineProperty(exports, "AttributesValues", {
  enumerable: true,
  get: function get() {
    return _AttributesValues.default;
  }
});
Object.defineProperty(exports, "InlineObject", {
  enumerable: true,
  get: function get() {
    return _InlineObject.default;
  }
});
Object.defineProperty(exports, "Item", {
  enumerable: true,
  get: function get() {
    return _Item.default;
  }
});
Object.defineProperty(exports, "ItemPictures", {
  enumerable: true,
  get: function get() {
    return _ItemPictures.default;
  }
});
Object.defineProperty(exports, "Variations", {
  enumerable: true,
  get: function get() {
    return _Variations.default;
  }
});
Object.defineProperty(exports, "VariationsAttributeCombinations", {
  enumerable: true,
  get: function get() {
    return _VariationsAttributeCombinations.default;
  }
});
Object.defineProperty(exports, "CategoriesApi", {
  enumerable: true,
  get: function get() {
    return _CategoriesApi.default;
  }
});
Object.defineProperty(exports, "ItemsApi", {
  enumerable: true,
  get: function get() {
    return _ItemsApi.default;
  }
});
Object.defineProperty(exports, "ItemsHealthApi", {
  enumerable: true,
  get: function get() {
    return _ItemsHealthApi.default;
  }
});
Object.defineProperty(exports, "OAuth20Api", {
  enumerable: true,
  get: function get() {
    return _OAuth20Api.default;
  }
});
Object.defineProperty(exports, "RestClientApi", {
  enumerable: true,
  get: function get() {
    return _RestClientApi.default;
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _Attributes = _interopRequireDefault(require("./model/Attributes"));

var _AttributesValueStruct = _interopRequireDefault(require("./model/AttributesValueStruct"));

var _AttributesValues = _interopRequireDefault(require("./model/AttributesValues"));

var _InlineObject = _interopRequireDefault(require("./model/InlineObject"));

var _Item = _interopRequireDefault(require("./model/Item"));

var _ItemPictures = _interopRequireDefault(require("./model/ItemPictures"));

var _Variations = _interopRequireDefault(require("./model/Variations"));

var _VariationsAttributeCombinations = _interopRequireDefault(require("./model/VariationsAttributeCombinations"));

var _CategoriesApi = _interopRequireDefault(require("./meli-marketplace-lib/CategoriesApi"));

var _ItemsApi = _interopRequireDefault(require("./meli-marketplace-lib/ItemsApi"));

var _ItemsHealthApi = _interopRequireDefault(require("./meli-marketplace-lib/ItemsHealthApi"));

var _OAuth20Api = _interopRequireDefault(require("./meli-marketplace-lib/OAuth20Api"));

var _RestClientApi = _interopRequireDefault(require("./meli-marketplace-lib/RestClientApi"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
