// meli-sdk-node/build/lib/types/items.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MeliItemAttributesEnum = void 0;
var MeliItemAttributesEnum;
(function (MeliItemAttributesEnum) {
    MeliItemAttributesEnum["id"] = "id";
    MeliItemAttributesEnum["site_id"] = "site_id";
    MeliItemAttributesEnum["title"] = "title";
    MeliItemAttributesEnum["subtitle"] = "subtitle";
    MeliItemAttributesEnum["seller_id"] = "seller_id";
    MeliItemAttributesEnum["category_id"] = "category_id";
    MeliItemAttributesEnum["official_store_id"] = "official_store_id";
    MeliItemAttributesEnum["price"] = "price";
    MeliItemAttributesEnum["base_price"] = "base_price";
    MeliItemAttributesEnum["original_price"] = "original_price";
    MeliItemAttributesEnum["currency_id"] = "currency_id";
    MeliItemAttributesEnum["initial_quantity"] = "initial_quantity";
    MeliItemAttributesEnum["available_quantity"] = "available_quantity";
    MeliItemAttributesEnum["sold_quantity"] = "sold_quantity";
    MeliItemAttributesEnum["sale_terms"] = "sale_terms";
    MeliItemAttributesEnum["buying_mode"] = "buying_mode";
    MeliItemAttributesEnum["listing_type_id"] = "listing_type_id";
    MeliItemAttributesEnum["start_time"] = "start_time";
    MeliItemAttributesEnum["stop_time"] = "stop_time";
    MeliItemAttributesEnum["end_time"] = "end_time";
    MeliItemAttributesEnum["condition"] = "condition";
    MeliItemAttributesEnum["permalink"] = "permalink";
    MeliItemAttributesEnum["thumbnail_id"] = "thumbnail_id";
    MeliItemAttributesEnum["thumbnail"] = "thumbnail";
    MeliItemAttributesEnum["secure_thumbnail"] = "secure_thumbnail";
    MeliItemAttributesEnum["pictures"] = "pictures";
    MeliItemAttributesEnum["video_id"] = "video_id";
    MeliItemAttributesEnum["accepts_mercadopago"] = "accepts_mercadopago";
    MeliItemAttributesEnum["shipping"] = "shipping";
    MeliItemAttributesEnum["seller_address"] = "seller_address";
    MeliItemAttributesEnum["seller_contact"] = "seller_contact";
    MeliItemAttributesEnum["location"] = "location";
    MeliItemAttributesEnum["coverage_areas"] = "coverage_areas";
    MeliItemAttributesEnum["attributes"] = "attributes";
    MeliItemAttributesEnum["warnings"] = "warnings";
    MeliItemAttributesEnum["status"] = "status";
    MeliItemAttributesEnum["sub_status"] = "sub_status";
    MeliItemAttributesEnum["tags"] = "tags";
    MeliItemAttributesEnum["warranty"] = "warranty";
    MeliItemAttributesEnum["catalog_product_id"] = "catalog_product_id";
    MeliItemAttributesEnum["domain_id"] = "domain_id";
    MeliItemAttributesEnum["date_created"] = "date_created";
    MeliItemAttributesEnum["last_updated"] = "last_updated";
    MeliItemAttributesEnum["health"] = "health";
    MeliItemAttributesEnum["catalog_listing"] = "catalog_listing";
    MeliItemAttributesEnum["channels"] = "channels";
})(MeliItemAttributesEnum = exports.MeliItemAttributesEnum || (exports.MeliItemAttributesEnum = {}));
