// gmelisdk/src/libs/services/authentication.service.ts
import { request } from "../httpRequest"
import { endPoints} from "../api/endpoints"
import { restmeli } from "../interfaces/authentication.interface"

export const modelAuthMeli = (code:string|undefined  ) =>{
    return {
        grant_type: 'authorization_code',
        client_id: process.env.MERCADOLIBRE_APIKEY,
        client_secret: process.env.MERCADOLIBRE_SECRETKEY,
        code: code,
        redirect_uri: process.env.REDIRECT_URI
    }
}

export const modelAuthRefreshToken=(refresh_token:string )=>{
    return {
        grant_type: 'refresh_token',
        client_id: process.env.MERCADOLIBRE_APIKEY,
        client_secret: process.env.MERCADOLIBRE_SECRETKEY,
        redirect_uri: process.env.REDIRECT_URI,
        refresh_token:refresh_token
    }
}

export const authenticateByCode = async (code:string):Promise<any> =>{
    const params =  modelAuthMeli(code );
    console.log(params);
    
    const options = {
        method:"POST",
        headers: {
            "Content-Type":"application/json"
        },
        body:JSON.stringify(params)
    }
    return  await request( endPoints.oauth_url, options )
}

export const authenticateByRefreshToken = async (refresh_token:string|undefined ) =>{
    const params =  modelAuthRefreshToken(refresh_token );
    const options = {
        method:"POST",
        headers: {
            "Accept": "application/json",
            "Content-Type":"application/x-www-form-urlencoded"
        },
        body:JSON.stringify(params)
    }
    return   await request( endPoints.oauth_url, options )
}
